### DISCUSSION TITLE
WordPress + nginx + MariaDB

### DISCUSSION THREAD

**Discussion by iWaoc** [leokhoa/laragon#488] [Q&A]
Hi, I am a beginner using Laragon and WordPress.

How can I run WordPress locally with nginx and MariaDB in Laragon?

When I configure Laragon to install WordPress with Apache and MariaDB using "Quick App", everything works fine, but when I want to use nginx, the browser returns "404 Not Found nginx/1.24.0".

Is there any step by step guide to solve this problem or make it work?

Thanks in advance.

**Comment by robisadev** [comment]
In Laragon Preferences > Services & Ports, change your Nginx port to 80 or use `http://mysite:8080/`

**Reply by iWaoc** [reply]
Hola, Gracias por tu ayuda.

Lo que me has indicado, no funcionó, ya que ni siquiera estaba activo o funcionando nginx.

En mi caso, luego de investigar mucho, encontré que debía configurar en el archivo "auto.[proyectname].conf" el siguiente código:

```
location / {
        try_files $uri $uri/ /index.php$is_args$args;
        autoindex on;
    }
    
    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass php_upstream;
        #fastcgi_pass unix:/run/php/php7.0-fpm.sock;
    }

    location ~ ^/files/(.*)$ {
        try_files /wp-content/blogs.dir/$blogid/$uri /wp-includes/ms-files.php?file=$1;
        access_log off;
        log_not_found off;
        expires max;
    }

    location ~ ^/[_0-9a-zA-Z-]+/files/(.*)$ {
        try_files /wp-content/blogs.dir/$blogid/$uri /wp-includes/ms-files.php?file=$1;
        access_log off;
        log_not_found off;
        expires max;
    }
```


Con ello y otras configuraciones propias de nginx, fue que pude resolver el problema.
